### ISSUE TITLE
Error while updating config or settings

### ISSUE THREAD

**Issue by axcl** [gocd/gocd#9405]
##### Issue Type

- Bug Report

##### Summary

Whenever i tries to change config for roles or any plugins it throws following error
I have also tried completing removing and adding plugin again but somehow config still stucked.

```
Someone has modified the entity. Please update your copy with the changes and try again.

Someone has modified the configuration for role 'Admin'. Please update your copy of the config with the changes and try again.
```

###### Basic environment details

* Go Version: 21.2.0
* JAVA Version: 15.0.2
* OS: Linux 4.15.0-144-generic
* Browser vendor and version (if relevant): Latest chrome

##### Steps to Reproduce

1. 
2. 
3. 
4. 

##### Any other info

https://ibb.co/7yswJwK
https://ibb.co/2S0Qksd

**Comment by msharbaji** [comment]
I am facing the same issue, anyone faces the issue and manage to solve it?

**Comment by msharbaji** [comment]
@axcl have you fixed the issue?

**Comment by axcl** [comment]
No i have switched to Jenkins

**Comment by chadlwilson** [comment]
A similar problem was reported at #9964. Were you folks using GoCD behind some sort of caching reverse proxy or CDN, e.g Cloudflare?

**Comment by msharbaji** [comment]
@chadlwilson Yes, indeed it has a caching reverse proxy which Cloudflare, my colleague reported the problem you mentioned on, We add page rule to respect the strong ETag, but I'd suggest if the GoCD can catch it and accept either weak ETag or strong ETag.

Thanks for your support, really helped us with this hint and unblock us with operational stuff :)

**Comment by chadlwilson** [comment]
No worries, and thank you for confirming!

I am glad to have gotten to the bottom of the problem. Now to figure out the right way to address it - will track in the linked ticket.

If you have any information on other APIs/screens you have seen this type of problem with (or ones where updates definitely worked without the page rule) that'd be helpful, as currently we dont quite understand why the issue has been reported against plugin settings and roles, since almost all the APIs that update a single "entity" or piece of config have the same protection to avoid lost updates between different users' API calls, so I'd have expected issues to be reported against lots of APIs.

If folks are mainly using things such as pipelines-as-YAML for non-admin functions, that might explain it, since I imagine role config and plugin settings are the two more common admin functions which are more likely to involve UI configuration.

**Comment by msharbaji** [comment]
@chadlwilson indeed not just the updating role, we had an issue with all updates, example: update pipeline template we were facing the same issue :)

**Comment by Uriyo** [comment]
indeed not just the updating role, we had an issue with all updates, example: update pipeline template we were facing the same issue :)
 
 i am also facing this issue even in the latest version

**Comment by chadlwilson** [comment]
@Uriyo Please open a new issue with a proper explanation of your problem/environment along with what you have checked already. Also look at #9964 as well.

If your proxy messes with headers or does not propagate them properly, you can have similar problems, regardless of what GoCD does.

**Comment by Uriyo** [comment]
ok sure @chadlwilson
